lib/OktaAuthBase.ts

```diff
@@ -60,7 +62,7 @@ export default class OktaAuthBase implements OktaAuth, SigninAPI {
   }
 
   // { username, password, (relayState), (context) }
-  signIn(opts: SigninOptions): Promise<AuthTransaction> {
+  signIn(opts: SigninOptions | SigninWithRedirectOptions): Promise<AuthTransaction> | Promise<void> {
```

Comment:
maybe we can rename "SigninOptions" so its more clear. "SigninWithCredentialsOptions" ?

Reply:
I think we would need to leave at least an alias for "SigninOptions" though to avoid a breaking change.

Reply:
Should we consider types for backward compatibility? That's the only reason why I kept the name.

Reply:
Yes we will need to keep type names for compatibility. We can rename the type then add an alias with a comment that it may be deprecated in a later version.